src/Components/OATGraphViewer/OATGraphViewer.tsx

```diff
@@ -131,85 +131,100 @@ const OATGraphViewer = ({ state, dispatch }: OATGraphProps) => {
 
     useEffect(() => {
         // Detects when a Model is deleted outside of the component and Updates the elements state
-        const importModelsList = [];
+        let importModelsList = [...elements];
         if (importModels.length > 0) {
             importModels.forEach((input) => {
                 const node = elements.find(
                     (element) => element.id === input['@id']
                 );
-                if (!node) {
-                    let relationships = [];
-                    let contents = [];
-                    input['contents'].forEach((content) => {
-                        if (content['@type'] === OATComponentHandleName) {
-                            const componentRelationship = new ElementEdge(
+                if (node) {
+                    importModelsList = importModelsList.reduce(
```

Comment:
Can you help me understand what's happening in this reduce function? If we're filtering, then `filter` might be a better option?